### PR TITLE
Add subscription mechanism using async iterators

### PR DESCRIPTION
**Pull request by lorenzofox3**
The follow up of the previous pull request rebaseing the code base

### CODE REVIEW

**Comment by lorenzofox3** [comment]
Note: users should probably be warned that this (anti)pattern would eventually lead to memory leak:
event emitter is pushed based whereas async iterator is pull based so if the subscriber is slower to process than the emitter to emit the internal queue will keep growing.

File: readme.md [hunk 1]
```diff
@@ -68,8 +68,13 @@ emitter.once('🦄').then(data => {
 
 emitter.emit('🦄', '🌈');
 ```
+#### events(eventName)
 
-#### emit(eventName, data?)
+Get an asynchronous iterator which buffers data each time an event is emitted.
```
Comment (sindresorhus) [inline readme.md]: Can you add a code example of using it?

File: readme.md [hunk 2]
```diff
@@ -93,6 +98,12 @@ Returns a method to unsubscribe.
 
 Remove an `onAny` subscription.
 
+#### anyEvent()
+
+Get an async iterator which buffers a tuple of an event name and data each time an event is emitted.
+
+Call `return()` on the iterator to remove the subscription.
```
Comment (sindresorhus) [inline readme.md]: Can you add a usage example here too?

**Comment by sindresorhus** [comment]
You need to update https://github.com/sindresorhus/emittery/blob/master/index.d.ts too (The text should be the same as the readme, including examples).

**Comment by sindresorhus** [comment]
> You need to update /index.d.ts@master  too (The text should be the same as the readme, including examples).

@lorenzofox3 :arrow_up: :)

**Comment by lorenzofox3** [comment]
oops my bad I had not seen that one :)

**Comment by sindresorhus** [comment]
Thanks for working on this, @lorenzofox3 🙌

**Pull request merged**